File: go/two-fer/two_fer.go
package twofer

// ShareWith 
// -param: name a string (default: "you")
// -return: "One for " `name` ", one for me."
func ShareWith(name string) string { 
	// why can't I have a default value for function arguments? 
	if len(name) == 0 { // how do I check if nil? is it `?name`
		name = "you"
	}
	return "One for " + name + ", one for me."
}

File: java/two-fer/src/main/java/Twofer.java
public class Twofer {
  
  /**
   * This method takes in a String `name`
   * and returns "One for `name`, one for me."
   * if no `name` is given it is replaced by "you"
   * @param name a String
   * @return a formated String "One for `name`, one for me."
   */
  public String twofer(String name)
  {
    return String.format("One for %s, one for me."
    , (name == null || name.isEmpty()) ? "you" : name
    );
  }
}
